source/common/http/http1/codec_impl.cc

```diff
@@ -411,8 +411,9 @@ int ServerConnectionImpl::onHeadersComplete(HeaderMapImplPtr&& headers) {
         http_parser_pause(&parser_, 1);
       }
 
-    } else if (parser_.method == http_method::HTTP_POST ||
-               parser_.method == http_method::HTTP_PUT) {
+    } else if ((parser_.method == http_method::HTTP_POST ||
+                parser_.method == http_method::HTTP_PUT) &&
+               headers->ContentLength() == nullptr) {
```

Comment:
On server-side we don't support half closed. On client side we will correctly process response is client closes. Do you know what happens with current code for HTTP/1.1 if you POST without content-length? I'm actually not sure. Might be good to run your test without the fix to see what happens.

Ultimately in thinking about I agree this change makes sense, I would just update the comments a bit if you don't mind. (I'm just petrified of changes like this and I should have been more careful when the first one went by. We always wind up breaking some badly written app when we do something like this).